homeassistant/components/here_weather/config_flow.py

```diff
@@ -0,0 +1,151 @@
+"""Config flow for here_weather integration."""
+import logging
+from typing import Optional
+
+import herepy
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_API_KEY,
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_MODE,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+    CONF_UNIT_SYSTEM,
+    CONF_UNIT_SYSTEM_IMPERIAL,
+    CONF_UNIT_SYSTEM_METRIC,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_MODES,
+    DEFAULT_MODE,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    HERE_API_KEYS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_validate_user_input(hass: HomeAssistant, user_input: dict) -> None:
+    """Validate the user_input containing coordinates."""
+    await async_validate_name(hass, user_input)
+    here_client = herepy.DestinationWeatherApi(user_input[CONF_API_KEY])
+    await hass.async_add_executor_job(
+        here_client.weather_for_coordinates,
+        user_input[CONF_LATITUDE],
+        user_input[CONF_LONGITUDE],
+        herepy.WeatherProductType[user_input[CONF_MODE]],
+    )
+
+
+async def async_validate_name(hass: HomeAssistant, user_input: dict) -> None:
```

Comment:
Seems wild to not allow a duplicate name. I would remove this check.

Reply:
Removed